graylog2-web-interface/src/views/hooks/SyncWithQueryParameters.js

```diff
@@ -0,0 +1,53 @@
+// @flow strict
+import { useEffect } from 'react';
+import URI from 'urijs';
+import history from 'util/History';
+
+import { ViewStore } from 'views/stores/ViewStore';
+import View from 'views/logic/views/View';
+import { QueriesActions } from 'views/actions/QueriesActions';
+
+const useActionListeners = (actions, callback, dependencies) => {
+  useEffect(() => {
+    const unsubscribes = actions.map(action => action.listen(callback));
+    return () => unsubscribes.forEach(unsubscribe => unsubscribe());
+  }, dependencies);
+};
+
+const extractTimerangeParams = (timerange) => {
+  const { type } = timerange;
+  const result = { rangetype: type };
+
+  switch (type) {
+    case 'relative': return Object.entries({ ...result, relative: timerange.range });
+    case 'keyword': return Object.entries({ ...result, keyword: timerange.keyword });
+    case 'absolute': return Object.entries({ ...result, from: timerange.from, to: timerange.to });
+    default: return Object.entries(result);
+  }
+};
+
+export const syncWithQueryParameters = (query: string) => {
+  const { view } = ViewStore.getInitialState() || {};
+  if (view && view.type === View.Type.Search) {
+    const firstQuery = view.search.queries.first();
+    if (firstQuery) {
```

Comment:
Out of interest, is there a reason you are checking for the first query here, but take it for granted in `BindSearchParamsFromQuery.js`?

Reply:
Not really. Added the check to `SyncWithQueryParameters` too.